### PR TITLE
Ch8385

### PR DESCRIPTION
**Pull request by adamsdarlingtower**
Adds unit tests to sql expression, and fixs some edge case bugs that became visible while writing tests.

Does not test allocate()

### CODE REVIEW

**Review by rad-pat — APPROVED** [review]
Looks ok, but I'd prefer each test were smaller, or broken into sub tests

File: plaidcloud/utilities/tests/test_sql_expression.py [hunk 1]
```diff
@@ -0,0 +1,1368 @@
...
+
+    def test_get_from_clause(self):
+        # hoo boy
+        source_column_configs = [
+            {'source': 'Column1', 'dtype': 'text'},
+            {'source': 'Column2', 'dtype': 'numeric'},
+        ]
+        table = se.get_table_rep('table_12345', source_column_configs, 'anlz_schema')
+
+        # Should always return a Label object
```
[the remark this answers is not in the document]
Comment (adamsdarlingtower) [reply plaidcloud/utilities/tests/test_sql_expression.py:355]: Yup that makes sense
